aws-ssm-maintenancewindow/src/main/java/software/amazon/ssm/maintenancewindow/ReadHandler.java

```diff
@@ -1,27 +1,90 @@
+import software.amazon.cloudformation.exceptions.BaseHandlerException;
+import software.amazon.cloudformation.proxy.ProgressEvent;
 import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
 import software.amazon.cloudformation.proxy.Logger;
-import software.amazon.cloudformation.proxy.ProgressEvent;
-import software.amazon.cloudformation.proxy.OperationStatus;
 import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.ssm.maintenancewindow.translator.ExceptionTranslator;
+import software.amazon.ssm.maintenancewindow.translator.resourcemodel.GetMaintenanceWindowTranslator;
+import software.amazon.ssm.maintenancewindow.util.ClientBuilder;
 
 public class ReadHandler extends BaseHandler<CallbackContext> {
 
+    private static final SsmClient SSM_CLIENT = ClientBuilder.getClient();
+
+    private final GetMaintenanceWindowTranslator getMaintenanceWindowTranslator;
+    private final ExceptionTranslator exceptionTranslator;
+
+    ReadHandler() {
+        this.getMaintenanceWindowTranslator = new GetMaintenanceWindowTranslator();
+        this.exceptionTranslator = new ExceptionTranslator();
+    }
+
+    /**
+     * Used for unit tests.
+     *
+     * @param getMaintenanceWindowTranslator Translates GetMaintenanceWindow into ResourceModel objects.
+     * @param exceptionTranslator Translates service model exceptions.
+     */
+    ReadHandler(final GetMaintenanceWindowTranslator getMaintenanceWindowTranslator, final ExceptionTranslator exceptionTranslator) {
+        this.getMaintenanceWindowTranslator = getMaintenanceWindowTranslator;
+        this.exceptionTranslator = exceptionTranslator;
+    }
+
     @Override
     public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-        final AmazonWebServicesClientProxy proxy,
-        final ResourceHandlerRequest<ResourceModel> request,
-        final CallbackContext callbackContext,
-        final Logger logger) {
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final Logger logger) {
+
+        logger.log(String.format("Processing ReadHandler request %s", request));
 
         final ResourceModel model = request.getDesiredResourceState();
 
-        // TODO : put your code here
+        final ProgressEvent<ResourceModel, CallbackContext> progressEvent = new ProgressEvent<>();
+        progressEvent.setStatus(OperationStatus.FAILED);
+
+        final String windowId = model.getWindowId();
+
+        if (StringUtils.isNullOrEmpty(windowId)) {
+            progressEvent.setErrorCode(HandlerErrorCode.InvalidRequest);
+            progressEvent.setMessage("WindowId must be present to read the existing association.");
```

Comment:
nit: WindowId must be present to read the existing maintenance window

Reply:
Addressed!!!